client/main.lua

```diff
@@ -31,10 +37,17 @@ RegisterCommand('vol', function(source, args)
 	end
 end)
 
-function toggleVoice(tgtId, enabled)
+function toggleVoice(tgtId, enabled, effectType)
+	if enabled and effectType then
+		if effectType == 'radio' then
```

Comment:
why not do `if effectType and enabled and effectType == 'radio'`
It would allow for early escaping of the if check if its theres no effect type (small performance gain) along side not wasting another if statement and doing it all in one.

Reply:
because support for more effect types, such as 'phone'. we havent added it to our server yet because we couldnt get other effects working

Reply:
I guess I've never been a big fan of the if, elseif stuff, lua can somewhat do 'cases' in a weird way, but it looks a lot nicer imo

![image](https://user-images.githubusercontent.com/13681939/105649677-c50c4680-5e76-11eb-84b5-a0a910d565cf.png)

Reply:
The performance on that will be much worse. and its the same native? its just the argument `radioEffectId` that you change anyways

Reply:
Actually it's surprisingly not, it usually performs about the same, sometimes better/worse because of standard deviation (98-100ms). It is also easier to build off of in the future vs a chain of if/elseifs + better readability.

Heres the code I use if you want to try too (just change to the table call to a if statement.
```lua

-- radio submix
local radioEffectId = CreateAudioSubmix('Radio')
SetAudioSubmixEffectRadioFx(radioEffectId, 0)
SetAudioSubmixEffectParamInt(radioEffectId, 0, GetHashKey('default'), 1)
AddAudioSubmixOutput(radioEffectId, 0)

local submixFunctions = {
	['radio'] = function(tgtId)
		MumbleSetSubmixForServerId(tgtId, radioEffectId)
	end,
	['phone'] = function(tgtId)

	end
}

function toggleVoice(tgtId, enabled, submixType)
	if GetConvarInt('voice_enableRadioSubmix', 0) == 1 then
		if submixType then
			submixFunctions[submixType](tgtId)
		elseif not submixType then
			MumbleSetSubmixForServerId(tgtId, -1)
		end
	end

	MumbleSetVolumeOverrideByServerId(tgtId, enabled and volume or -1.0)
end

-- don't want constantly making a string be a part of the performance test as it would increase the standard deviation
local source = '10'
CreateThread(function()
	for i = 1, 100000 do
		toggleVoice(source, true, 'radio')
	end
end)
```

Reply:
Also, using Lua 5.4 drops the times by 20ms 👀

Reply:
Yes we're also adding lua 5.4 support, especally a better math.random was needed, but it has some issues with our code